crates/ruff_linter/src/rules/refurb/rules/reimplemented_operator.rs

```diff
@@ -78,11 +91,15 @@ impl Violation for ReimplementedOperator {
 
 /// FURB118
 pub(crate) fn reimplemented_operator(checker: &mut Checker, target: &FunctionLike) {
-    // Ignore methods.
-    if target.kind() == FunctionLikeKind::Function {
-        if checker.semantic().current_scope().kind.is_class() {
-            return;
-        }
+    // Ignore methods, whether defined using the `def` keyword or via a `lambda` assignment.
+    if checker.semantic().current_scope().kind.is_class()
+        && (target.is_function_def()
+            || checker
+                .semantic()
+                .current_statements()
+                .any(|stmt| matches!(stmt, Stmt::AnnAssign(_) | Stmt::Assign(_))))
+    {
+        return;
     }
 
```

Comment:
A comment explaining why we're testing for an assignment might be helpful. 

I wonder if we could avoid the false positive by testing if there's a parent expression. If that's the case, then the operator can't be the assignment's value?

Reply:
> I wonder if we could avoid the false positive by testing if there's a parent expression. If that's the case, then the operator can't be the assignment's value?

I don't think that's the case -- a lot of functions take an input function, do something with the function (monkey-patch an attribute onto the function, or store the function in a global registry somewhere, etc.), then return the same function. That's a very common pattern for decorators, for example:
- https://github.com/python/cpython/blob/3a77980002845c22e5b294ca47a12d62bf5baf53/Lib/typing.py#L2706-L2739
- https://github.com/python/cpython/blob/3a77980002845c22e5b294ca47a12d62bf5baf53/Lib/typing.py#L3738-L3770
- https://github.com/python/cpython/blob/3a77980002845c22e5b294ca47a12d62bf5baf53/Lib/typing.py#L2582-L2615

So all of these are still valid method definitions, and it would be a false positive if we flagged them with FURB118:

```py
from typing import final, override, no_type_check

class Foo:
    a = final(lambda self, other: self == other)
    b = override(lambda self, other: self == other)
    c = no_type_check(lambda self, other: self == other)
    d = final(override(no_type_check(lambda self, other: self == other)))
```

I'll add more tests involving constructs like these

Reply:
One heuristic that might work would be if we still emitted diagnostics for lambdas in `key=` arguments, since lambdas are very often used as `key` functions and it's very likely that the lambda will be thrown away if it's passed as a `key` function. That would avoid the false negative on the `zerver` codebase.

Complex heuristics like that make the rule harder to understand for users, though, so there's a tradeoff there. I think I'd prefer to avoid it for now.

Reply:
Oh, okay, that's indeed complicated. I'm fine keeping it as is.